### DISCUSSION TITLE
Structure of projects and components

### DISCUSSION THREAD

**Discussion by joubu** [WeblateOrg/weblate#9556] [Q&A]
Hello,
I am one of the maintainer of the Koha project (koha-community.org) and we would like to migrate to Weblate (from Pootle).
We are very excited with the changes and all the great features it will bring to our translators!
However I am having some questions about the implementation and how we should structure our projects and components.

We have basically 2 projects: a main one "koha" and one for the doc "koha-manual". For each of them we support different versions, and the translations are split into several po files.
For instance (for the manual):
locales/es/LC_MESSAGES/acquisitions.po
locales/es/LC_MESSAGES/administration.po
locales/fr/LC_MESSAGES/acquisitions.po
locales/fr/LC_MESSAGES/administration.po
There are 29 po's per language.

In Pootle we currently have the version at the project level, then the files listed per language:
koha-manual-23.05 > LANG > list of files (acquisitions, administration, etc.)

If I understand correctly, with Weblate we will need to have the version at the component level (to propagate changes between versions), and we cannot have several files per component anyway.

So we would have:
koha-manual > 23.05-acquisitions > LANG
koha-manual > 23.05-administration > LANG
...

Which means a lot of components, and not grouped by version.

Is this how it is supposed to work or am I missing something?

**Comment by joubu** [comment]
I guess "component lists" could help us to group them.

**Comment by nijel** [comment]
If you can wait one month, there will be a way to structure components in a project, see https://github.com/WeblateOrg/weblate/issues/263

**Comment by joubu** [comment]
That's awesome, I am looking forward to try it out! Thanks a lot for your work!

**Comment by joubu** [comment]
Thanks, @nijel for your work on this. It works nicely.

I have a question however. I have the following structure:
project / branch / component, where branch is the category (the idea was to group the component of the same branch).
When I create a second component and I link it with the first one using the weblate:// repo, it does not accept "weblate://project/branch/first_component", but only "weblate://project/first_component".
As I am not able to select a repository branch I feel like I am doing something wrong.

When I will create the second category for the other branch, I will need to point to a different repository branch (I don't mind having 1 repo per branch copied locally).

Is there a concept I am missing here?

**Reply by joubu** [reply]
I have opened a separate discussion - https://github.com/WeblateOrg/weblate/discussions/10112
